src/mod/endpoints/mod_sofia/mod_sofia.c

```diff
@@ -5092,9 +5098,37 @@ static switch_call_cause_t sofia_outgoing_channel(switch_core_session_t *session
 
 	if (session) {
 		o_channel = switch_core_session_get_channel(session);
+		caller_tech_pvt = switch_core_session_get_private(session);
+		// TEL-6085:
+		if (sofia_test_media_flag(caller_tech_pvt->profile, SCMF_RTP_TIMEOUT_SEC)) {
+			switch_channel_set_variable_printf(nchannel, "rtp_timeout_sec", "%d", caller_tech_pvt->profile->rtp_timeout_sec);
```

Comment:
Instead of setting a channel variable. Create a new variable in sofia private where we can pass the flag or value from the inbound profile. Add a condition in this [line](https://github.com/team-telnyx/freeswitch/blob/telnyx/telephony/master/src/mod/endpoints/mod_sofia/sofia_glue.c#L188) that the sofia private variable will take precedence over profile variable

Reply:
It can also be a channel variable if you prefer but it has to be different than `rtp_timeout_sec`. If that channel variable exist then same condition above, it take precedence over the profile variable. I'll let you decide which is better

Reply:
Then apply the same logic to the rest of the variable

Reply:
I have reworked the PR, please re-review :) thanks